### PR TITLE
Upgrade to v2.0.0 of the AzureAD Terraform Provider

### PR DESCRIPTION
**Pull request by stack72**
Fixes: #92
Fixes: #174
Fixes: #181
Fixes: #182﻿

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,11 @@ CHANGELOG
 =========
 
 ## HEAD (Unreleased)
+* Upgrade to v2.0.0 of the AzureAD Terraform Provider
+  Please Note:
+  * `azuread.ApplicationOAuth2Permission` and `azuread.azuread_application_oauth2_permission_scope` have been removed.
```
Comment (mikhailshilkov) [inline CHANGELOG.md:7]:
```suggestion
  * `azuread.ApplicationOAuth2Permission` and `azuread.ApplicationOAuth2PermissionScope` have been removed.  
```

File: provider/resources.go [hunk 2]
```diff
@@ -98,28 +98,27 @@ func Provider() tfbridge.ProviderInfo {
...
 		Resources: map[string]*tfbridge.ResourceInfo{
-			"azuread_application":                         {Tok: makeResource(mainMod, "Application")},
-			"azuread_application_password":                {Tok: makeResource(mainMod, "ApplicationPassword")},
-			"azuread_group":                               {Tok: makeResource(mainMod, "Group")},
-			"azuread_service_principal":                   {Tok: makeResource(mainMod, "ServicePrincipal")},
-			"azuread_service_principal_password":          {Tok: makeResource(mainMod, "ServicePrincipalPassword")},
-			"azuread_user":                                {Tok: makeResource(mainMod, "User")},
-			"azuread_group_member":                        {Tok: makeResource(mainMod, "GroupMember")},
-			"azuread_application_certificate":             {Tok: makeResource(mainMod, "ApplicationCertificate")},
-			"azuread_application_app_role":                {Tok: makeResource(mainMod, "ApplicationAppRole")},
```
Comment (mikhailshilkov) [inline provider/resources.go:109]: Do we need to changelog this removal too?

**Review by mikhailshilkov — APPROVED** [review]
LGTM modulo lint failure + two suggestions on the changelog

File: provider/resources.go (hunk 2 above)
Comment (stack72) [reply provider/resources.go:109]: Yes! I will fix that up :) Thank you! Great catch

**Pull request merged**